From ae62b9ce875c797f1ab0718bd827ef1c682fc318 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Tue, 24 Jan 2017 08:45:35 +0000
Subject: [PATCH] refpolicy: fix auditd fails to start

Reference sources: selinux-policy-3.13.1-225.6.fc25.src.rpm
Changes come from: policy-f25-base.patch

Fix avc denials:

  avc: denied { getcap } for pid=849 comm="auditctl" \
  scontext=system_u:system_r:auditctl_t:s0-s15:c0.c1023 \
  tcontext=system_u:system_r:auditctl_t:s0-s15:c0.c1023 \
  tclass=process

  avc: denied { setattr } for pid=848 comm="auditd" \
  name="audit" dev="tmpfs" ino=9569 \
  scontext=system_u:system_r:auditd_t:s15:c0.c1023 \
  tcontext=system_u:object_r:auditd_log_t:s15:c0.c1023 \
  tclass=dir

  avc: denied { search } for pid=731 comm="auditd" \
  name="/" dev="tmpfs" ino=9399 \
  scontext=system_u:system_r:auditd_t:s15:c0.c1023 \
  tcontext=system_u:object_r:tmpfs_t:s0 tclass=dir

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/system/logging.te | 3 +++
 1 file changed, 3 insertions(+)

--- refpolicy.orig/policy/modules/system/logging.te
+++ refpolicy/policy/modules/system/logging.te
@@ -102,6 +102,7 @@ ifdef(`enable_mls',`
 allow auditctl_t self:capability { fsetid dac_read_search dac_override };
 allow auditctl_t self:process getcap;
 allow auditctl_t self:netlink_audit_socket nlmsg_readpriv;
+allow auditctl_t self:process getcap;
 
 read_files_pattern(auditctl_t, auditd_etc_t, auditd_etc_t)
 allow auditctl_t auditd_etc_t:dir list_dir_perms;
@@ -150,6 +151,7 @@ allow auditd_t self:tcp_socket create_st
 allow auditd_t auditd_etc_t:dir list_dir_perms;
 allow auditd_t auditd_etc_t:file read_file_perms;
 
+manage_dirs_pattern(auditd_t, auditd_log_t, auditd_log_t)
 manage_files_pattern(auditd_t, auditd_log_t, auditd_log_t)
 allow auditd_t auditd_log_t:dir setattr;
 manage_lnk_files_pattern(auditd_t, auditd_log_t, auditd_log_t)
@@ -170,6 +172,7 @@ dev_read_sysfs(auditd_t)
 fs_getattr_all_fs(auditd_t)
 fs_search_auto_mountpoints(auditd_t)
 fs_rw_anon_inodefs_files(auditd_t)
+fs_search_tmpfs(auditd_t)
 
 selinux_search_fs(auditctl_t)
 
